### PR TITLE
mgr/cephadm: lock multithreaded access to OSDRemovalQueue

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Since the set can be changed also from the CLI thread

Fixes: https://tracker.ceph.com/issues/47700

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

Replaces: #37492

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by mgfritch — APPROVED** [review]
lgtm!

File: src/pybind/mgr/cephadm/services/osd.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import json
 import logging
+from threading import Lock
```
Comment (jmolmo) [inline src/pybind/mgr/cephadm/services/osd.py:3]: Why not to use RLock?

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/services/osd.py:3]: just cause I didn't needed to.

**Review by jmolmo — APPROVED** [review]
Not sure if the use of Rlock can speed up things ...  but LGTM.

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-01-13_11:19:08-rados:cephadm-wip-swagner3-testing-2021-01-12-1316-distro-basic-smithi/

Lots of failures, but they seem to be unrelated to this PR:
* https://tracker.ceph.com/issues/48873
* https://tracker.ceph.com/issues/48715
* https://tracker.ceph.com/issues/48142
* https://github.com/ceph/ceph/pull/38844#issuecomment-760089828

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-01-15_09:42:49-rados:cephadm-wip-swagner-testing-2021-01-14-1551-distro-basic-smithi/

* https://tracker.ceph.com/issues/48894
* https://tracker.ceph.com/issues/48715
* https://tracker.ceph.com/issues/48142

**Pull request merged**